source/iface/usm_allocations.rst

```diff
@@ -94,6 +94,199 @@ the ``device`` and ``context``.
 
 .. seealso:: `C++ Allocator <https://en.cppreference.com/w/cpp/named_req/Allocator>`__
 
+
+.. _allocation_functions :
+
+====================
+Allocation functions
+====================
+
+=======================
+``sycl::malloc_device``
+=======================
+
+::
+
+   void* sycl::malloc_device(size_t numBytes, const device& syclDevice,
+                          const context& syclContext,
+                          const property_list& propList = {});
+
+   template <typename T>
+   T* sycl::malloc_device(size_t count, const device& syclDevice,
+                          const context& syclContext,
+                          const property_list& propList = {});
+
+   void* sycl::malloc_device(size_t numBytes, const queue& syclQueue,
+                          const property_list& propList = {});
+
+   template <typename T>
+   T* sycl::malloc_device(size_t count, const queue& syclQueue,
+                          const property_list& propList = {});
+
+   void* sycl::aligned_alloc_device(size_t alignment, size_t numBytes,
+                          const device& syclDevice,
+                          const context& syclContext,
+                          const property_list& propList = {});
+
+   template <typename T>
+   T* sycl::aligned_alloc_device(size_t alignment, size_t count,
+                          const device& syclDevice,
+                          const context& syclContext,
+                          const property_list& propList = {});
+
+   void* sycl::aligned_alloc_device(size_t alignment, size_t numBytes,
+                          const queue& syclQueue,
+                          const property_list& propList = {});
+
+   template <typename T>
+   T* sycl::aligned_alloc_device(size_t alignment, size_t count,
+                          const queue& syclQueue,
+                          const property_list& propList = {})
+
+.. rubric:: Parameters
+
+==================  ===
+``alignment``       alignment of allocated data
+``numBytes``        allocation size in bytes
+``count``           number of elements
+``syclDevice``      See :ref:`device`
+``syclQueue``       See :ref:`queue`
+``syclContext``     See :ref:`context`
+``propList``
+==================  ===
+
+On success USM device allocation, these functions return a pointer to the newly
+allocated memory, which must eventually be deallocated with ``sycl::free`` in
+order to avoid a memory leak. If there are not enough resources to allocate the
+requested memory, these functions return ``nullptr``.
+
+When the allocation size is zero bytes (``numBytes`` or ``count`` is zero),
```

Comment:
Please replicate this paragraph in the host and shared allocation descriptions as well.  It's a SYCL 2020 spec bug that this only appeared in the device allocation section - I'm fixing that in the SYCL spec, so please replicate the paragraph on "When the allocation size is zero bytes..." in all three malloc sections to match the SYCL spec fix.

Reply:
Followed instructions from https://github.com/KhronosGroup/SYCL-Docs/pull/485  and added fix in commit https://github.com/KhronosGroup/SYCL-Docs/pull/485